File: LeetCode Problems/PlusOne Code/PlusOne.py
class Solution:
    def plusOne(self, digits: list[int]) -> list[int]:
        """
        digits: integer array digits
        number: incremented array
        """
        # join the list to form an integer
        digits = int("".join(list(map(str, digits))))
        # Increment and again turn it to a list
        number = list(str(digits + 1))
        # Return the incremented list
        return number

File: LeetCode Problems/PlusOne Code/Plusone.cpp
class Solution {
public:
vector plusOne(vector& digits) {

   for(int i = digits.size()-1;i>=0;i--){
       
       if(digits[i] < 9){
           digits[i]++;
           break;
       }
       else if(i != 0 && digits[i]==9){
           digits[i] = 0;
       }
       else{
         digits[0]=0;
         digits.insert(digits.begin(),1);
       }
   }
    return digits;
}
};
